priority-queue-sf/PriorityQueue.Shared/QueueManager.cs

```diff
@@ -90,38 +98,37 @@ await namespaceManager.CreateTopicAsync(this.topicName)
                     .ConfigureAwait(false))
                 {
                     // Setup the filter for the subscription based on the priority.
-                    var ruleDescription = new RuleDescription
+                    var ruleDescription = new CreateRuleOptions
                     {
                         Name = "PriorityFilter",
-                        Filter = new SqlFilter($"Priority = '{priority}'")
+                        Filter = new SqlRuleFilter($"Priority = '{priority}'")
                     };
 
+                    var options = new CreateSubscriptionOptions(this.topicName, subscription);
+
                     try
                     {
-                        await namespaceManager.CreateSubscriptionAsync(this.topicName, subscription, ruleDescription)
-                            .ConfigureAwait(false);
+                        await namespaceManager.CreateSubscriptionAsync(options, ruleDescription).ConfigureAwait(false);
                     }
-                    catch (MessagingEntityAlreadyExistsException)
+                    catch (ServiceBusException ex) when (ex.Reason == ServiceBusFailureReason.MessagingEntityAlreadyExists)
                     {
                         Trace.TraceInformation($"Messaging entity already created: {subscription}");
                     }
-                    catch (MessagingException ex) when (((ex.InnerException as WebException)?.Response as HttpWebResponse)?.StatusCode == HttpStatusCode.Conflict)
+                    catch (ServiceBusException ex) when (((ex.InnerException as RequestFailedException)?.Status) == 409)
```

Comment:
minor: HttpStatusCode.Conflict instead of 409?

Reply:
Updated. Cast the Enum.